bftengine/src/bftengine/ReplicaConfigSerializer.cpp

```diff
@@ -117,14 +117,12 @@ void ReplicaConfigSerializer::serializeDataMembers(ostream &outStream) const {
   // Serialize replicaPrivateKey
   serializeKey(config_->replicaPrivateKey, outStream);
 
-  serializePointer(config_->thresholdSignerForSlowPathCommit, outStream);
-  serializePointer(config_->thresholdVerifierForSlowPathCommit, outStream);
-
-  serializePointer(config_->thresholdSignerForCommit, outStream);
-  serializePointer(config_->thresholdVerifierForCommit, outStream);
-
-  serializePointer(config_->thresholdSignerForOptimisticCommit, outStream);
-  serializePointer(config_->thresholdVerifierForOptimisticCommit, outStream);
+  // Serialize threshold crypto system
+  concord::serialize::Serializable::serialize(outStream, config_->thresholdSystemType_);
+  concord::serialize::Serializable::serialize(outStream, config_->thresholdSystemSubType_);
```

Comment:
A lot of `serialize` we should rename it :)

Reply:
> Thank you for the great PR, I think it makes the code base much cleaner.
> In my opinion, if a singleton does not conform with the unit tests, the singleton has to be replaced with something else, not the unit tests. For example, `CryptoManager` can be passed as a constant reference.

This unittest always was a problematic one because of the way it was written. I don't think it's essential and needed to be maintained.